lib/build.py

```diff
@@ -450,6 +468,10 @@ def prepare(self, miss_ct):
          Typically, subclasses will set up enough state for self.sid_input to
          be valid, then call super().prepare().
 
+         WARNING: Instructions that modify image metadata (at this writing,
+         ARG ENV FROM LABEL SHELL WORKDIR) must do so here, not in execute(),
+         so that metadata is available to late instructions even on cache hit.
+
```

Comment:
Looks like there's a merge error here: This paragraph moved to item 3 below, so modify that instead.

Reply:
Good catch, thanks!